source/windows/mutex.c

```diff
@@ -0,0 +1,46 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/mutex.h>
+
+int aws_mutex_init(struct aws_mutex *mutex, struct aws_allocator *allocator) {
+    mutex->allocator = allocator;
+    mutex->mutex_handle.Ptr = 0;
```

Comment:
Where's the `InitializeSRWLock`? https://msdn.microsoft.com/en-us/library/windows/desktop/ms683483(v=vs.85).aspx

Reply:
you only call InitializeSRWLock on dynamically allocated memory. If it's statically allocated, it just gets initialized to zero.

"The InitializeSRWLock function is used to initialize a SRW lock dynamically. To initialize the structure statically, assign the constant SRWLOCK_INIT"

SRWLOCK_INIT is just {0}, but that doesn't compile so I manually did the init.

Reply:
You don't actually know if this is statically allocated or not. However, since this should be ABI-stable, leave a comment explaining that this is a faster version of `InitializeSRWLock`.

Reply:
The way I parse the meaning for "static" from the msdn docs, is it is a member of the struct as a value, not as a pointer to a different allocation.

Reply:
Static allocation just means you can allocate a SRW lock as a statically allocated structure without needing to coordinate an initialization call at startup. The documentation doesn't actually say that setting a heap SRW lock to 0 performs initialization, but the ABI implies it, so it'll have to work, at least until windows comes up with a new platform with a different initializer.